next.config.js

```diff
@@ -4,5 +4,6 @@ module.exports = {
   env: {
```

Comment:
I believe you are using this config property wrong. The point here is to set for example `NEXT_PUBLIC_YOUTUBE_API` to the value of your API key, like `la1kj5674lk72g52b3k4j2` or something. And, then you can use it like `process.env.NEXT_PUBLIC_FIREBASEAPI`.

But because you do not want to push your key to the repository I would recommend to remove `env` property from the config, and use `.env.local` for local development by adding `NEXT_PUBLIC_YOUTUBE_API` and `NEXT_PUBLIC_FIREBASEAPI` there, (`NEXT_PUBLIC_MONGODB_URI` could be used in `.env` if its not secret).

Reply:
Well, it may sound stupid but I did it the exact same way initially and it didn't work.
It used to return `undefined` and so I came up with that solution.

I tried it this time and it worked, surprisingly.
Thank you for correcting me here.